drivers/flash/spi_nor.c

```diff
@@ -213,7 +214,11 @@ static int spi_nor_read(struct device *dev, off_t addr, void *dest,
 err:
 	SYNC_UNLOCK();
 
-	return ((u8_t *)dest - p_dest);
+	if (((u8_t *)dest - p_dest) == size_req) {
```

Comment:
no need for the `if-else` when a return is added upon error returned by `spi_nor_cmd_addr_read`